cmd/iaviewer/main.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/tendermint/iavl"
+	dbm "github.com/tendermint/tendermint/libs/db"
+)
+
+// TODO: make this configurable?
+const (
+	DefaultCacheSize int = 10000
+)
+
+func main() {
+	args := os.Args[1:]
+	if len(args) < 2 || (args[0] != "data" && args[0] != "shape" && args[0] != "versions") {
+		fmt.Println("Usage: iaviewer <data|shape|versions> <leveldb dir> [version number]")
+		os.Exit(1)
+	}
+
+	version := 0
+	if len(args) == 3 {
+		var err error
+		version, err = strconv.Atoi(args[2])
+		if err != nil {
+			fmt.Printf("Invalid version number: %s\n", err)
+			os.Exit(3)
```

Comment:
Nit: Besides indicating an error, is it useful to exit with a different non-zero code per failure? Here 2, below 3? 
I would suggest to exit with `os.Exit(1)` in all failure cases.

Reply:
Okay, too verbose I guess.
I will normalize to os.Exit(1) for everything